video/video_decoder.cpp

```diff
@@ -112,6 +112,15 @@ bool VideoDecoder::needsUpdate() const {
 	return false;
 }
 
+void VideoDecoder::delayMillis(uint msecs) {
+	if (!needsUpdate())
+		g_system->delayMillis(MIN<uint>(msecs, getTimeToNextFrame()));
+#ifdef __DS__
+	else
+		g_system->delayMillis(0); /* This is needed to keep the mixer and timers active */
```

Comment:
Wouldn't be safer to put `1` here? IIRC some backends (`ATARI` included) use something like
```C
const uint32 threshold = getMillis() + msecs;
	while (getMillis() < threshold) {
		update();
	}
```
so `0` here wouldn't execute it in every case. Perhaps a better solution is possible here as well (like an addition to OSystem API to do exactly that, update the mixer and timers).

Reply:
I propose to use this instead:
```
const uint32 threshold = getMillis() + msecs;
do {
    update();
} while (getMillis() < threshold);
```
This makes sure the update is done once when calling delayMillis.